src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -105,6 +105,14 @@ function checkAndWarnForMutatedStyle(style1, style2, component) {
   if (shallowEqual(style1, style2)) {
     return;
   }
+  // Don't show mutated style warning if a value is NaN since the styles weren't mutated.
+  for (var key in style1) {
+    var value = style1[key];
+
+    if (isNaN(value) && ('' + value) === 'NaN') {
```

Comment:
I don't think the `&& ('' + value) === 'NaN'` is correct.  For instance, the string "NaN" would pass that check, I believe.

You probably want to use `Number.isNaN()` or one of the polyfills listed in the docs:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN